pkg/api/componentreadiness/component_report.go

```diff
@@ -608,6 +677,180 @@ func (s *sampleQueryGenerator) queryTestStatus() (crtype.ReportTestStatus, []err
 	return crtype.ReportTestStatus{SampleStatus: sampleStatus}, errs
 }
 
+type fallbackTestQueryReleasesGenerator struct {
+	client                     *bqcachedclient.Client
+	cacheOption                cache.RequestOptions
+	commonQuery                string
+	groupByQuery               string
+	queryParameters            []bigquery.QueryParameter
+	allJobVariants             crtype.JobVariants
+	BaseRelease                string
+	BaseStart                  time.Time
+	BaseEnd                    time.Time
+	CachedFallbackTestStatuses crtype.FallbackReleases
+}
+
+func (f *fallbackTestQueryReleasesGenerator) getTestFallbackReleases() (*crtype.FallbackReleases, []error) {
+	wg := sync.WaitGroup{}
+	fallbackRelease := f.BaseRelease
+	f.CachedFallbackTestStatuses = newFallbackReleases()
+
+	releases, errs := GetReleaseDatesFromBigQuery(f.client)
+
+	if errs != nil {
+		return nil, errs
+	}
+
+	var err error
+	// currently gets current base plus previous 3
+	// current base is just for testing but use could be
+	// extended to no longer require the base query
+	for i := 0; i < 4; i++ {
+
+		// currently fetching the baseRelease via fallback to
+		// perform validation of the test data
+		// this gets uncommented when ready
+		// and the section at the bottom removed
+		// fallbackRelease, err = previousRelease(fallbackRelease)
+		// if err != nil {
+		// 	log.WithError(err).Errorf("Failure determining fallback release for %s", fallbackRelease)
+		// 	continue
+		// }
+
+		start := f.BaseStart
+		end := f.BaseEnd
+		var crRelease *crtype.Release
+
+		for _, release := range releases {
+			if release.Release == fallbackRelease {
+				crRelease = &release
+				if release.GADate != nil && release.GA30DayPrior != nil {
+					end = *release.GADate
+					start = *release.GA30DayPrior
+				}
+				break
+			}
+		}
+
+		wg.Add(1)
+		go func(queryRelease *crtype.Release, queryStart, queryEnd time.Time) {
```

Comment:
This isn't the only case where we're not doing it, but we need to start propagating the context from the http Request down into component readiness and using it.  When a client disconnects, these still continue executing which isn't ideal.

At the minimum it makes sippy a target for a ddos attack (clients can quickly reconnect over and over again and spawn off many goroutines that stay running).

i.e.:

```golang
func (s *Server) jsonComponentReportFromBigQuery(w http.ResponseWriter, req *http.Request) {
    ctx := req.Context()

    // [....]
    componentreadiness.GetComponentReportFromBigQuery(ctx, ...)
    // [...]
}

func GetComponentReportFromBigQuery(ctx context.Context, ...) {
    // [...]
    go func() {
        select {
        case <-ctx.Done():
            return // terminate when context does
        default:
            // actual body of work to do
        }
    }()
    // [...]
}

```

Reply:
I filed a separate card for this: https://issues.redhat.com/browse/TRT-1845